File: src/Tema001/Tema001.py
def imprimir_linea(longitud=40, caracter='-'):
    print(caracter * longitud)

def imprimir_titulo(titulo):
    imprimir_linea(40, '=')
    print(titulo.upper())
    imprimir_linea(len(titulo))

def imprimir_subtitulo(subtitulo):
    imprimir_linea(len(subtitulo))
    print(subtitulo)
    imprimir_linea(len(subtitulo))

def main():
    imprimir_titulo("Primitivos")
    imprimir_subtitulo("Enteros")

    precio = 12
    otro_precio = precio
    print(precio == otro_precio)

    otro_precio = 24
    print(precio == otro_precio)

    otro_precio = 12
    print(precio == otro_precio)

    imprimir_subtitulo("Strings")

    cadena_texto = "Una cadena"
    otra_cadena = cadena_texto
    tercera_cadena = str("Una cadena")

    print(cadena_texto, otra_cadena, tercera_cadena)
    print(cadena_texto is otra_cadena)
    print(cadena_texto is tercera_cadena)

    cadena_texto = "XYZ"
    print(cadena_texto, otra_cadena, tercera_cadena)

    tercera_cadena = "ABC"
    print(cadena_texto, otra_cadena, tercera_cadena)

    imprimir_titulo("Listas")

    lista1 = [1, 2, 3]
    lista2 = lista1
    lista3 = [1, 2, 3]

    print("Las listas tienen el mismo contenido:", lista1 == lista3)
    print("Las listas son el mismo objeto:", lista1 is lista2)
    print("Las listas son diferentes objetos:", lista1 is not lista3)

    lista2[0] = 100
    print("Primer elemento de lista1:", lista1[0])

    imprimir_titulo("Objetos Personalizados")

    persona1 = Persona("Alice")
    persona2 = persona1
    persona3 = Persona("Alice")

    print("Los nombres son iguales:", persona1.nombre == persona3.nombre)
    print("Los objetos son el mismo objeto:", persona1 is persona2)
    print("Los objetos son diferentes objetos:", persona1 is not persona3)

    persona2.nombre = "Bob"
    print("Nombre de persona1:", persona1.nombre)

class Persona:
    def __init__(self, nombre):
        self.nombre = nombre

if __name__ == "__main__":
    main()

File: src/Tema001/Tema001.php
<?php

function imprimirLinea($longitud = 40, $caracter = '-') {
    echo str_repeat($caracter, $longitud) . PHP_EOL;
}

function imprimirTitulo($titulo) {
    imprimirLinea(40, '=');
    echo strtoupper($titulo) . PHP_EOL;
    imprimirLinea(strlen($titulo));
}

function imprimirSubtitulo($subtitulo) {
    imprimirLinea(strlen($subtitulo));
    echo $subtitulo . PHP_EOL;
    imprimirLinea(strlen($subtitulo));
}

imprimirTitulo("Primitivos");

imprimirSubtitulo("Números");

$precio = 12;
$otroPrecio = $precio;
echo ($precio === $otroPrecio ? "true" : "false") . PHP_EOL;

$otroPrecio = 24;
echo ($precio === $otroPrecio ? "true" : "false") . PHP_EOL;

$otroPrecio = 12;
echo ($precio === $otroPrecio ? "true" : "false") . PHP_EOL;

imprimirSubtitulo("Strings");

$cadenaTexto = "Una cadena";
$otraCadena = $cadenaTexto;
$terceraCadena = "Una cadena";

echo "$cadenaTexto - $otraCadena - $terceraCadena" . PHP_EOL;
echo ($cadenaTexto === $otraCadena ? "true" : "false") . PHP_EOL;
echo ($cadenaTexto === $terceraCadena ? "true" : "false") . PHP_EOL;

imprimirTitulo("Arrays");

$array1 = array(1, 2, 3);
$array2 = $array1;
$array3 = $array1;  

echo "Los arrays tienen el mismo contenido: " . (json_encode($array1) === json_encode($array3) ? "true" : "false") . PHP_EOL;
echo "Los arrays son el mismo objeto: " . ($array1 === $array2 ? "true" : "false") . PHP_EOL;
echo "Los arrays son diferentes objetos: " . ($array1 !== $array3 ? "true" : "false") . PHP_EOL;

$array2[0] = 100;
echo "Primer elemento de arr1: " . $array1[0] . PHP_EOL;

$array4 = [1, 2, 3];
$array5 = &$array4;
$array5[0] = 100;
echo "array4[0]: " . $array4[0] . "\n";  


imprimirTitulo("Objetos");

$objeto1 = (object) ['nombre' => "Alice"];
$objeto2 = $objeto1;
$objeto3 = clone $objeto1;  

echo "Los nombres son iguales: " . ($objeto1->nombre === $objeto3->nombre ? "true" : "false") . PHP_EOL;
echo "Los objetos son el mismo objeto: " . ($objeto1 === $objeto2 ? "true" : "false") . PHP_EOL;
echo "Los objetos son diferentes objetos: " . ($objeto1 !== $objeto3 ? "true" : "false") . PHP_EOL;

$objeto2->nombre = "Bob";
echo "Nombre de obj1: " . $objeto1->nombre . PHP_EOL;

?>
